### DISCUSSION TITLE
Shorter forms of {}-using statements

### DISCUSSION THREAD

**Discussion by HugoRoss** [dotnet/csharplang#2938] [Language Ideas]
I would find it helpful if variable declarations could be skipped in pre-8.0 `using` declarations with the (inferred) variable still accessible through a leading dot.

Current syntax:

    using (var myStream = File.OpenRead(filePath)) {
        return myStream.ReadByte();
    }

Suggested new syntax:

    using (File.OpenRead(filePath)) {
        return .ReadByte();
    }

(Suggestion of single line statements removed again, not practical)

(see also discussion of #2894)

**Comment by redradist** [comment]
WTF !!

Man, seriously =))))))))
Unfortunately it is very hard to read ...

It seems very similar with some Kotlin feature ...
If I would suggest something like this it would be with the following syntax:
```csharp
context (File.OpenRead(filePath))
{
    return ReadByte();
}
```
where context is a new keyword ;)

This code will transformed into:
```csharp
{
  var myStream = File.OpenRead(filePath);
  return myStream .ReadByte();
}
```

**Comment by HaloFour** [comment]
A `with` statement by any other name....

**Comment by redradist** [comment]
> 
> 
> A `with` statement by any other name....

@HaloFour Yeah, it is very good !! ;)

**Comment by agocke** [comment]
```C#
using (File.OpenRead(filePath)) {
...
}
```

This is already valid C# code and it does not mean what you want it to mean.
